df.world-site.xml

```diff
@@ -260,6 +260,29 @@
         <enum-item name='Town' comment='not hamlet'/>
     </enum-type>
 
+    <enum-type type-name='fortress_type' base-type='int16_t'>
+        <enum-item name='NIL' value='-1'/>
```

Comment:
Have you seen these set to -1? Doesn't hurt to include it for completeness, I suppose.
Also, I suggest using `None` to match other enums in this case.

Reply:
Yes, all of them are -1 in all cases where the site type isn't the one controlling the use of the enum (and I don't think I've seen them be -1 when the site is of the correct type).

I'll change change the names to None, except for the lair_type, where I'll use all uppercase to match the rest, and copy the names therehedwig used. I'll still define them as their own types outside of the struct, though, as it will get rather messy with 3 of them in there.

Reply:
If you want to make a dfhack pull request to fix the plugins that referenced these fields, I can tie the two pull requests together in BuildMaster.

Reply:
Thanks, I'll do that, but first I have to figure out why changing the enum value from NIL to NONE broke compilation. The file static.ctors.inc is deleted when I clean the project, but it still insists on using the removed enum value when it is regenerated, even though the .h file has been regenerated as well and contains the new data. There's some insidious cache somewhere that does something "smart", apparently...